src/BSON/Binary.c

```diff
@@ -69,6 +69,32 @@ static bool php_phongo_binary_init_from_hash(php_phongo_binary_t* intern, HashTa
 	return false;
 } /* }}} */
 
+static HashTable* php_phongo_binary_get_properties_hash(phongo_compat_object_handler_type* object, bool is_temp) /* {{{ */
+{
+	php_phongo_binary_t* intern;
+	HashTable*           props;
+
+	intern = Z_OBJ_BINARY(PHONGO_COMPAT_GET_OBJ(object));
+
+	PHONGO_GET_PROPERTY_HASH_INIT_PROPS(is_temp, intern, props, 2);
+
+	if (!intern->data) {
+		return props;
+	}
+
+	{
+		zval data, type;
+
+		ZVAL_STRINGL(&data, intern->data, intern->data_len);
+		zend_hash_str_update(props, "data", sizeof("data") - 1, &data);
+
+		ZVAL_LONG(&type, intern->type);
+		zend_hash_str_update(props, "type", sizeof("type") - 1, &type);
+	}
```

Comment:
I've wondered about this before. What's the practical reason for nesting? Is it to only conditionally allocate the two `zval` structs on the stack and avoid doing so if there's no data to handle?

Reply:
These aren't even conditionally allocated. We're just using braces to create a scope and allow us to declare variables inline. Our C compiler options would otherwise flag this as a warning and require that we declare these at the top of the function scope.